TheJumpCloud/jcapi-powershell
# SA-3169 Search-JcSdkCommandResult

## Issues
* [SA-3169](https://jumpcloud.atlassian.net/browse/SA-3169) - Search-JcSdkCommandResult

## What does this solve?

`Search-JcSdkCommandResult` added to module
`Get-JcSdkCommandResultByCommandId` added to module (this function was previously tied to `Get-JCCommandResult`)
By removing `Get-JcSdkCommandResultByCommandId` from `Get-JCCommandResult`, the ID parameter now works

## Is there anything particularly tricky?

NO

## How should this be tested?

Tests have been written to check the newly added functions/ functionality but the new functions can be tested:

`Get-JCSdkCommandResult` — Should return all command results in an org
`Get-JCSdkCommandResult -ID {SomeCommandResultID}` — Should return a single command results in an org
`Get-JCSdkCommandResultByCommandID -Id {SomeCommandID}` — Should return all command results in an org for a particular command
`Search-JCSdkCommandResult` should return all command results 
Try to filter command results with the `filter` `fields` parameters to limit the commands returned. 

## Screenshots

[SA-3169]: https://jumpcloud.atlassian.net/browse/SA-3169?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
